internal/cmd/root.go

```diff
@@ -151,6 +154,14 @@ func (c *rootCommand) execute() {
 	if errors.As(err, &ecerr) {
 		exitCode = int(ecerr.ExitCode())
 	}
+	var differentBinaryError runDifferentBinaryError
+
+	if errors.As(err, &differentBinaryError) {
+		err = differentBinaryError.customBinary.run(c.globalState)
```

Comment:
I find getting the custom binary from the error unintuitive. 
I would expect that the error was about missing dependencies and handle it here by attempting the binary provisioning.  That would have a better separation of concerns between the error and how it is handled.

Reply:
Aaah ,... sorry.

Do you mean like e48314ebf ?